app/server/models/user.py

```diff
@@ -513,14 +550,23 @@ def set_pin(self, supplied_pin=None, is_activated=False):
 
         self.hash_password(pin)
 
-    def is_resetting(self):
-        self.clear_expired_tokens()
+    def is_resetting_password(self):
+        self.clear_expired_password_reset_tokens()
         is_resetting = len(self.password_reset_tokens) > 0
         return is_resetting
 
-    # TODO(ussd): change to a field once we figure out what's the deal with resetting
-    def pin_failed_attempts(self):
-        return 0
+    def is_resetting_pin(self):
+        self.clear_expired_pin_reset_tokens()
+        is_resetting = len(self.pin_reset_tokens) > 0
+        return is_resetting
+
+    @property
+    def pin(self):
+        return self.pin_hash
+
+    @pin.setter
+    def pin(self, pin):
+        self.pin_hash = self.salt_hash_secret(pin)
```

Comment:
I think these make more sense to follow the same pattern as the password; a hashed password is set explicitly using `def hash_password(self, password):...` and the hash itself isn't really accessed directly at all,